### PR TITLE
Add data sharing

### PR DESCRIPTION
**Pull request by surchs**
Draft for data sharing docs

### CODE REVIEW

**Comment by surchs** [comment]
@isabelbacellar, do you maybe have some feedback on this?

**Comment by gdevenyi** [comment]
@surchs can you possibly use proper inline-references for the links and such? I think we might be able to auto-generate a "all links" appendix webpage if we go about doing it that way.

File: docs/conf.py [hunk 1]
```diff
@@ -65,6 +76,15 @@
...
 
+# Set some RTD theme config.  This includes the entire navigation structure
+# into the sidebar of all pages.  However, expanding the sections isn't
+# provided yet on the RTD theme (see
+# https://github.com/readthedocs/sphinx_rtd_theme/issues/455).
+html_theme_options = {
+    'collapse_navigation': False,
+    'navigation_depth': 3,
+}
+
```
Comment (gdevenyi) [inline docs/conf.py:87]: I was working on this and I thought I added it somewhere hrmmm

File: docs/conf.py [hunk 2]
```diff
@@ -34,9 +34,20 @@
     "myst_parser",
     "sphinx.ext.githubpages",
     "sphinx_rtd_dark_mode",
-    "sphinx.ext.autosectionlabel"
+    "sphinx.ext.autosectionlabel",
+    "sphinx.ext.todo"
```
Comment (gdevenyi) [inline docs/conf.py:38]: This is cool, how does it work?

**Review by gdevenyi — CHANGES_REQUESTED** [review]

File: docs/conf.py (hunk 2 above)
Comment (surchs) [reply docs/conf.py:38]: It allows you to generate the docs with TODO boxes and to keep TODO boxes in the text as reminders of future fixes: https://www.sphinx-doc.org/en/master/usage/extensions/todo.html

File: docs/conf.py (hunk 1 above)
Comment (surchs) [reply docs/conf.py:87]: Do you mean you had already pushed this? I didn't see a conflict before my PR

**Comment by surchs** [comment]
> @surchs can you possibly use proper inline-references for the links and such? I think we might be able to auto-generate a "all links" appendix webpage if we go about doing it that way.

Hey @gdevenyi, can you explain what you mean by this? Or comment inside of a file where the link is not used the way you suggest? Not sure exactly what the difference is you are referring to

**Comment by gdevenyi** [comment]
https://myst-parser.readthedocs.io/en/latest/syntax/syntax.html#footnotes

**Comment by surchs** [comment]
Those are footnotes, I'm not using them. I use links to link to outside resources. Sphinx also has a glossary for term definitions. Are you saying the outside resources should be footnotes?

**Comment by gdevenyi** [comment]
Right they call it that but that mixes two concepts together in typography.

I was hoping there might be a plugin that we can use to aggregate all external links into a seperate resources page.

I haven't searched deeply enough to confirm.

**Comment by surchs** [comment]
Yeah I see. That might be possible, I haven't seen it yet. Sphinx by default uses RST and there are some syntax mixings going on here because MyST is markdown. So maybe there are additional options to be discovered. 

But for now, is there anything you want to change about the way markdown links are set in this PR?

**Comment by isabelbacellar** [comment]
> @isabelbacellar, do you maybe have some feedback on this?

@surchs amazing work! 🏆 Just a few general comments:

1. "Although making your work accessible to others involves costs (effort and time, mostly), it considerably increases the reach of your work and thus also benefits your academic career." >> Say that this process will be a lot smoother if you plan for sharing from day 1 (i.e. keep your data structured and organized, figure our ethics/DMP beforehand etc). I think that's an important point because part of the DNP's goal is to help people manage well their data on a daily basis.

2. "Your first step should be to ensure that you can in fact share the raw or derivative data you intend to share. Here are some reasons you may be constrained in your ability to share your data:" >> I would just note that "constraints" don't necessarily mean abandoning data sharing plans, but sometimes just adapting to what's possible (e.g. sharing less/part, use anonymization methods etc).

3. Perhaps warn that sharing data through journals may create copyright issues. 

4. Finally, if you're not covering it elsewhere, maybe mention from a technical point of view how using the platform may make data sharing easier (e.g. something about transferring the data, if applicable).

**Comment by gdevenyi** [comment]
@surchs 
> But for now, is there anything you want to change about the way markdown links are set in this PR?

After a bunch of searching it seems the only solution right now otherwise is to go to full citations, which is way too much work, so the current format is fine :+1:

**Comment by surchs** [comment]
Thanks @isabelbacellar for the feedback!

> "Although making your work accessible to others involves costs (effort and time, mostly), it considerably increases the reach of your work and thus also benefits your academic career." >> Say that this process will be a lot smoother if you plan for sharing from day 1 (i.e. keep your data structured and organized, figure our ethics/DMP beforehand etc). I think that's an important point because part of the DNP's goal is to help people manage well their data on a daily basis.

> "Your first step should be to ensure that you can in fact share the raw or derivative data you intend to share. Here are some reasons you may be constrained in your ability to share your data:" >> I would just note that "constraints" don't necessarily mean abandoning data sharing plans,Finally, if you're not covering it elsewhere, maybe mention from a technical point of view how using the platform may make data sharing easier (e.g. something about transferring the data, if applicable). but sometimes just adapting to what's possible (e.g. sharing less/part, use anonymization methods etc).
EDIT: or restricting access to shared datasets (e.g. requiring users to be PIs)

I tried to address the first two points in the commits. Does that answer your points?

> Perhaps warn that sharing data through journals may create copyright issues.

This is a good idea. Do you have a good source that I can link to here, I am not intimately familiar with these concerns.

> Finally, if you're not covering it elsewhere, maybe mention from a technical point of view how using the platform may make data sharing easier (e.g. something about transferring the data, if applicable).

I think this is an important point that we should cover somewhere. At the moment, the technical solutions we offer revolve around the ability to store data, not how you store it. I.e.: we give people space on a managed hard-drive but don't tell them how to use it. The solutions we offer for research methods such as data sharing are your and our expertise on the matter, documentation, and training. But they are not technical solutions or mandates. Gabe has set up some documents to discuss these points in the next meeting (I'll be in Europe then).

@all-contributors please add @isabelbacellar for review

**Comment by surchs** [comment]
@gdevenyi thanks are you OK with merging this as is then?

@all-contributors please add @gdevenyi for review

**Comment by isabelbacellar** [comment]
> Thanks @isabelbacellar for the feedback!
> 
> > "Although making your work accessible to others involves costs (effort and time, mostly), it considerably increases the reach of your work and thus also benefits your academic career." >> Say that this process will be a lot smoother if you plan for sharing from day 1 (i.e. keep your data structured and organized, figure our ethics/DMP beforehand etc). I think that's an important point because part of the DNP's goal is to help people manage well their data on a daily basis.
> 
> > "Your first step should be to ensure that you can in fact share the raw or derivative data you intend to share. Here are some reasons you may be constrained in your ability to share your data:" >> I would just note that "constraints" don't necessarily mean abandoning data sharing plans,Finally, if you're not covering it elsewhere, maybe mention from a technical point of view how using the platform may make data sharing easier (e.g. something about transferring the data, if applicable). but sometimes just adapting to what's possible (e.g. sharing less/part, use anonymization methods etc).
> > EDIT: or restricting access to shared datasets (e.g. requiring users to be PIs)
> 
> I tried to address the first two points in the commits. Does that answer your points?
> 
> > Perhaps warn that sharing data through journals may create copyright issues.
> 
> This is a good idea. Do you have a good source that I can link to here, I am not intimately familiar with these concerns.
> 
> > Finally, if you're not covering it elsewhere, maybe mention from a technical point of view how using the platform may make data sharing easier (e.g. something about transferring the data, if applicable).
> 
> I think this is an important point that we should cover somewhere. At the moment, the technical solutions we offer revolve around the ability to store data, not how you store it. I.e.: we give people space on a managed hard-drive but don't tell them how to use it. The solutions we offer for research methods such as data sharing are your and our expertise on the matter, documentation, and training. But they are not technical solutions or mandates. Gabe has set up some documents to discuss these points in the next meeting (I'll be in Europe then).
> 
> @all-contributors please add @isabelbacellar for review

@surchs for 1+2, I think it's good. When can add more details about 2 (anonymization etc.) in the future, perhaps creating a dedicated page for that. But for now (i.e. a general data sharing page), I think it's enough. As for 3, I learned about that in [this presentation](https://msss365.sharepoint.com/:b:/r/teams/GroupedesoutienCentrederecherche/Shared%20Documents/Open%20Science/OPEN%20SCIENCE/3.06%20OS%20support%20tools%20and%20services%20(and%20case%20studies)/The%20Future%20is%20Open%20-%20Becoming%20an%20Open%20Researcher.pdf?csf=1&web=1&e=CcOTRa), slide 47. During the webinar they said that this applied for data shared through supplementary materials. Finally, as for point 4, we can surely discuss it later and also update it later, if we don't have any solutions for now.

**Comment by surchs** [comment]
Thanks @isabelbacellar!
I think the bot is confused here!

File: docs/conf.py (hunk 1 above)
Comment (gdevenyi) [reply docs/conf.py:87]: I thought I had but I can't find anything so :+1:

File: docs/research_methods/sharing_data.md [hunk 3]
```diff
@@ -0,0 +1,261 @@
+# Sharing data
+
+Publishing research data is an important step in ensuring that your research outputs are following
+the [**F**indable **A**ccessible **I**nteroperable **R**eusable principles](https://www.go-fair.org/fair-principles/)
```
Comment (gdevenyi) [inline docs/research_methods/sharing_data.md:4]: I think maybe have the acronym by itself as well.

File: docs/research_methods/sharing_data.md [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+Generally, 
+there are two options:
+
+1. You are sharing __a dataset__ that you have acquired (e.g. raw data) or that you have processed (derived data).
+Your goal here is that someone else might want to find these data useful, will reuse them and give you credit for your work.
+2. You are sharing __data as part of an empirical manuscript submission__. 
+Your goal here is to support your findings and figures with the data that support your analyses - 
+and possibly to fulfill the requirements of the journal.
+
+The main difference between these two options is that option 2. is a specification of 1.
```
Comment (gdevenyi) [inline docs/research_methods/sharing_data.md:85]: > specification

I'm not sure that's the right word here, but I'm not sure what you're trying to say

File: docs/research_methods/sharing_data.md [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+  - specific pre-processing, inclusion or exclusion steps
+  - outlier analyses
+  - etc
+- What are the steps taken between your raw data and the major outcomes you want to show your readers
+  - Here it can be helpful to look at your methods figure
+  - Alternatively, draw a flow chart on a piece of paper or write it down as hierarchical text
+  - If you do share code with your paper, each step should ideally map to a specific command or script
+  - For each step, make a note of the input and output data required
+- What intermediate data do you want to share?
+  - Some analyses steps take too long or require too many resources for a reader to reasonably re-run them
```
Comment (gdevenyi) [inline docs/research_methods/sharing_data.md:177]: Or be too large?

**Review by gdevenyi — CHANGES_REQUESTED** [review]

**Comment by gdevenyi** [comment]
I added a few small comments, otherwise good to go :+1:

File: docs/research_methods/sharing_data.md (hunk 4 above)
Comment (isabelbacellar) [reply docs/research_methods/sharing_data.md:85]: special case?

**Comment by surchs** [comment]
OK, thanks both! Incorporated your feedback. I'll go ahead and merge this now!

**Comment by surchs** [comment]
@all-contributors please add @isabelbacellar for review

**Comment by isabelbacellar** [comment]
Great work! I'm super happy with this page!!

Just a few small things:

1. "a requirement by funding agencies **and if your group**. If your group"" >> remove bold part.
2. "Slides by McGill on data sharing constraint considerations" > those slides are from the FRQ.
3. Section "To prepare your data for sharing, you should:" >> some items are in lower case, but not others. Also some items are actionable steps and others are links. It breaks the flow of the text a little bit. Perhaps write "Additional resources:" or "Check/See:" before the links?

**Comment by surchs** [comment]
Ah, thanks @isabelbacellar, good catch. 
Do you maybe want to attempt a pull request to fix these? No problem if not, I can get to that when I'm back

**Comment by isabelbacellar** [comment]
@surchs I definitely would like to learn, but could we leave it for next time? :) I have to get back to a few other tasks still today

**Comment by gdevenyi** [comment]
@isabelbacellar you can go here:
https://github.com/DouglasNeuroInformatics/Douglas_User_Documentation/edit/master/docs/research_methods/sharing_data.md

and perform your suggested edits on the document, and then at the bottom select the second option "Create a new branch for this commit and start a pull request." which will kick off this process for your proposed changes for us to discuss.

**Comment by gdevenyi** [comment]
WOW. Vs-code in a browser window. Mind. Blown.

**Pull request merged**